checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,371 @@
+   */
+  static class CheckerInputFormat
+      extends InputFormat<IntWritable, NullWritable> {
+    public static boolean sCreateDone = false;
+
+    /**
+     * An input split consisting of an integer 1.
+     */
+    static class CheckerInputSplit extends InputSplit implements Writable {
+      int mValue;
+
+      public CheckerInputSplit() { }
+
+      public CheckerInputSplit(int inputValue) {
+        mValue = inputValue;
+      }
+
+      @Override
+      public long getLength() throws IOException {
+        return 0;
+      }
+
+      @Override
+      public String[] getLocations() throws IOException {
+        return new String[]{};
+      }
+
+      @Override
+      public void readFields(DataInput in) throws IOException {
+        mValue = WritableUtils.readVInt(in);
+      }
+
+      @Override
+      public void write(DataOutput out) throws IOException {
+        WritableUtils.writeVInt(out, mValue);
+      }
+    }
+
+    /**
+     * A record reader converting input to record-oriented view.
+     */
+    static class CheckerRecordReader
+        extends RecordReader<IntWritable, NullWritable> {
+      int mIntegerVal;
+      IntWritable mKey = null;
+
+      public CheckerRecordReader() {
+      }
+
+      @Override
+      public void initialize(InputSplit split, TaskAttemptContext context) {
+        mIntegerVal = ((CheckerInputSplit) split).mValue;
+      }
+
+      @Override
+      public void close() {
+        // NOTHING
+      }
+
+      @Override
+      public IntWritable getCurrentKey() {
+        return mKey;
+      }
+
+      @Override
+      public NullWritable getCurrentValue() {
+        return NullWritable.get();
+      }
+
+      @Override
+      public float getProgress() {
+        if (sCreateDone) {
+          return 1.0f;
+        } else {
+          return 0.0f;
+        }
+      }
+
+      @Override
+      public boolean nextKeyValue() {
+        if (mKey == null) {
+          mKey = new IntWritable(mIntegerVal);
+        }
+        if (sCreateDone) {
+          return false;
+        } else {
+          sCreateDone = true;
+          return true;
+        }
+      }
+    }
+
+    @Override
+    public RecordReader<IntWritable, NullWritable> createRecordReader(InputSplit split,
+        TaskAttemptContext context) {
+      return new CheckerRecordReader();
+    }
+
+    /**
+     * Creates the desired number of splits.
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) {
+      int numSplits = job.getConfiguration().getInt(MRJobConfig.NUM_MAPS, 1);
+      List<InputSplit> splits = new ArrayList<>();
+      for (int split = 0; split < numSplits; split++) {
+        splits.add(new CheckerInputSplit(1));
+      }
+      return splits;
+    }
+  }
+
+  /**
+   * In each mapper node, we will check whether this node can recognize
+   * Alluxio classes and filesystem.
+   */
+  protected static class CheckStatusMapper extends Mapper<IntWritable, NullWritable, Text, Text> {
```

Comment:
do we need a `IntWritable` as input key? can we also use `NullWritable` for key?

Reply:
FYI, I am not sure the answer:)